packages/extension/src/newtab/App.tsx

```diff
@@ -71,17 +81,26 @@ function InternalApp({
   const { contentScriptGranted } = useContentScriptStatus();
   const routeChangedCallbackRef = useTrackPageView();
 
+  const { user, isAuthReady } = useAuthContext();
+  const { growthbook } = useGrowthBookContext();
+  const isPageReady =
+    (growthbook?.ready && router?.isReady && isAuthReady) || isTesting;
+  const onboardingV2 = useFeature(feature.onboardingV2);
+
+  const shouldRedirectOnboarding =
+    !user && isPageReady && onboardingV2 !== OnboardingV2.Control && !isTesting;
+
   useQuery(EXTENSION_PERMISSION_KEY, () => ({
     requestContentScripts,
     registerBrowserContentScripts,
     getContentScriptPermission,
   }));
 
   useEffect(() => {
-    if (routeChangedCallbackRef.current) {
+    if (routeChangedCallbackRef.current && isPageReady) {
       routeChangedCallbackRef.current();
     }
-  }, [routeChangedCallbackRef]);
+  }, [isPageReady, routeChangedCallbackRef]);
```

Comment:
This will slow down our new tab experience because growthbook does not initialize inline? Not sure what we can do with existing setup but might be critical since we want newtab to be fast?

Reply:
Ah actually, it does not really delay, seen it wrong where `isPageReady` is used, it's just for tracking.

Reply:
Yeah, this is just for tracking. And if it turns out it slows it down, I think I can remove it and then just accept that we get sent an extra analytics event for `/` page.

Reply:
Yeah, I was more worried for actual display/rendering. Should be ok.